proxy/src/lib.rs

```diff
@@ -348,7 +346,7 @@ where
 
 fn serve<R, B, E, F, G>(
     bound_port: BoundPort,
-    tls_config: tls::ServerConfigWatch,
+    tls_config: Option<(tls::Identity, tls::ServerConfigWatch)>,
```

Comment:
TIOLI: Since the `(tls::Identity, tls::ServerConfigWatch)` tuple shows up in a couple of places, maybe it would be clearer as a struct?

Reply:
It is a good idea to create a struct for this. I think that refactoring would be easier to do after this and the client part is merged. I will check now.

Reply:
Okay, that sounds good to me. Let's do that in a future PR, then.